frost-core/src/benches.rs

```diff
@@ -0,0 +1,201 @@
+                    let (_nonce, _commitment) = frost::round1::commit(
+                        participant_identifier,
+                        key_packages
+                            .get(&participant_identifier)
+                            .unwrap()
+                            .secret_share(),
+                        rng,
+                    );
+                })
+            },
+        );
+
+        let mut nonces: HashMap<_, _> = HashMap::new();
+        let mut commitments: HashMap<_, _> = HashMap::new();
+
+        for participant_index in 1..(min_signers + 1) {
+            let participant_identifier = participant_index.try_into().expect("should be nonzero");
+            let (nonce, commitment) = frost::round1::commit(
+                participant_identifier,
+                key_packages
+                    .get(&participant_identifier)
+                    .unwrap()
+                    .secret_share(),
+                rng,
+            );
+            nonces.insert(participant_identifier, nonce);
+            commitments.insert(participant_identifier, commitment);
+        }
+
+        let message = "message to sign".as_bytes();
+        let comms = commitments.clone().into_values().collect();
+        let signing_package = frost::SigningPackage::new(comms, message.to_vec());
+
+        group.bench_with_input(
+            BenchmarkId::new("Round 2", min_signers),
+            &(
+                key_packages.clone(),
+                nonces.clone(),
+                signing_package.clone(),
+            ),
+            |b, (key_packages, nonces, signing_package)| {
+                b.iter(|| {
+                    let participant_identifier = 1u16.try_into().expect("should be nonzero");
+                    let key_package = key_packages.get(&participant_identifier).unwrap();
+                    let nonces_to_use = &nonces.get(&participant_identifier).unwrap();
+                    let _signature_share =
+                        frost::round2::sign(signing_package, nonces_to_use, key_package).unwrap();
+                })
+            },
+        );
+
+        let mut signature_shares = Vec::new();
+        for participant_identifier in nonces.keys() {
+            let key_package = key_packages.get(participant_identifier).unwrap();
+            let nonces_to_use = &nonces.get(participant_identifier).unwrap();
+            let signature_share =
+                frost::round2::sign(&signing_package, nonces_to_use, key_package).unwrap();
+            signature_shares.push(signature_share);
+        }
+
+        group.bench_with_input(
+            BenchmarkId::new("Aggregate", min_signers),
+            &(signing_package.clone(), signature_shares.clone(), pubkeys),
+            |b, (signing_package, signature_shares, pubkeys)| {
+                b.iter(|| {
+                    let _group_signature_ =
+                        frost::aggregate(signing_package, &signature_shares[..], pubkeys).unwrap();
```

Comment:
###### Optional simplification
```suggestion
                    frost::aggregate(signing_package, &signature_shares[..], pubkeys).unwrap();
```